### PR TITLE
fix flaky TestEventLoop and TestSubscriber

### LINKED ISSUE #154: tests: TestEventLoop is flaky
In #153 tests failed once from:
```
=== RUN   TestEventLoop
    event_test.go:38: 
        	Error Trace:	event_test.go:38
        	Error:      	Received unexpected error:
        	            	failure to retrieve data square: merkledag: not found
        	Test:       	TestEventLoop
```

### LINKED ISSUE #173: tests: `TestSubscriber` can hang / sometimes fails due to `context deadline exceeded`
## Proof
image

## Refs
Found in #171 

## Possible solution
There has to be a small 200 milliseconds timeout after the first Subscription on PubSub Topic. Basically, we need to ensure that peers notified each other about joining a topic before they are sending the msg. Otherwise, msg will never find a recipient.

### PR DESCRIPTION
**Pull request by Wondertan**
Closes #154 
Closes #173

### CODE REVIEW

File: service/block/event_test.go [hunk 1]
```diff
@@ -31,9 +32,9 @@ func TestEventLoop(t *testing.T) {
 	numBlocks := 3
 	expectedBlocks := mockFetcher.generateBlocks(t, numBlocks)
 
-	ctx, cancel = context.WithCancel(context.Background())
-	t.Cleanup(cancel)
-
+	// as we calculate extended square twice, this may take a lot of time, causing the main thread expect the result
+	// earlier than its ready.
+	time.Sleep(100 * time.Millisecond)
```
Comment (adlerjohn) [inline service/block/event_test.go:37]: This seems a bit...brittle? Instead of sleeping, shouldn't there be some sort of async `await` or callback?

Comment (Wondertan) [reply service/block/event_test.go:37]: Well, we just shouldn't calculate extended headers twice ;) But that's a whole different story...
Adding callbacks and/or awaiting on chan to API just to make test happier is not a good idea, though I understand where you are coming from.

**Comment by renaynay** [comment]
Looks like it's still broken though :( @Wondertan

**Comment by Wondertan** [comment]
@renaynay, All the checks successful

**Review by renaynay — APPROVED** [review]
LGTM thank you

File: service/block/event_test.go (hunk 1 above)
Comment (liamsi) [reply service/block/event_test.go:37]: Agree with John on this: I think this kind of timeouts are the reason we experience non-deterministic test-flakiness in tendermint. No need to change anything here though. And as you said the square will only be computed once in the future (if we have an issue for that I would recommend linking this comment or code there).

File: service/header/subscription_test.go [hunk 2]
```diff
@@ -17,33 +18,36 @@ func TestSubscriber(t *testing.T) {
...
 		pubsub.WithMessageSignaturePolicy(pubsub.StrictNoSign))
 	require.NoError(t, err)
-	topic, err := gossub2.Join(ExtendedHeaderSubTopic)
+
+	// create header Service
+	headerServ2 := NewHeaderService(nil, nil, pubsub2)
+	err = headerServ2.Start(context.Background())
 	require.NoError(t, err)
 
+	time.Sleep(2 * time.Second)
```
Comment (liamsi) [inline service/header/subscription_test.go:44]: Similar as above: https://github.com/celestiaorg/celestia-node/pull/186/files#r746171777

**Review by liamsi — APPROVED** [review]

File: service/block/event_test.go (hunk 1 above)
Comment (Wondertan) [reply service/block/event_test.go:37]: @renaynay, do we have an issue for this? I feel like you created one at some point?

Comment (renaynay) [reply service/block/event_test.go:37]: i think this https://github.com/celestiaorg/celestia-node/issues/125

File: service/header/subscription_test.go (hunk 2 above)
Comment (renaynay) [reply service/header/subscription_test.go:44]: https://github.com/celestiaorg/celestia-node/issues/125

Comment (Wondertan) [reply service/header/subscription_test.go:44]: Note, this is not fixable due to the way pubsub(gossipbsub) works. Pubsub's test themselves are using 2 second timeout. Only in the recent release a new feature appeared to wait here, but it is also brittle. For ref https://github.com/libp2p/go-libp2p-pubsub/pull/452.

**Review by renaynay — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**